pandas/tseries/index.py

```diff
@@ -1769,13 +1782,34 @@ def indexer_between_time(self, start_time, end_time, include_start=True,
         -------
         values_between_time : TimeSeries
         """
-        from dateutil.parser import parse
+        def _parse_time_string(time_object):
+        
```

Comment:
can u factor this routine out and out it in tseries/tools.py (eg next to the datetime parsing ones)

thrn just call it from here (u can also move the these to the corresponding testing module)